dbms/include/DB/Functions/FunctionsURL.h

```diff
@@ -996,6 +971,47 @@ struct CutSubstringImpl
 };
 
 
+struct UrlDecodeImpl
+{
+	static void vector(const ColumnString::Chars_t & data, const ColumnString::Offsets_t & offsets,
+		ColumnString::Chars_t & res_data, ColumnString::Offsets_t & res_offsets)
+	{
+		res_data.reserve(data.size());
+		size_t size = offsets.size();
+		res_offsets.resize(size);
+
+		size_t prev_offset = 0;
+		size_t res_offset = 0;
+
+		for (size_t i = 0; i < size; ++i)
+		{
+			const char * current = reinterpret_cast<const char *>(&data[prev_offset]);
+			std::string url = decodeUrl(StringView(current, offsets[i] - prev_offset - 1));
```

Comment:
> Another way is changing the interface of the decodeUrl so that it can accept destination string as second parameter.

That's Ok.

PS. Function must be named `decodeURL`, not `decodeUrl`.